Update and cross-reference migration notes. (#1686)
opendatacube/datacube-core: docs/installation/MIGRATION-1.8-to-1.9.rst

   The handling of lineage in the legacy postgres index driver has not changed - the postgres driver does NOT support
   the new lineage API.

6. Support for multi-dimensional loading of hyperspectral datasets

>    Use of this feature currently requires installing `odc-stac`_.
> 
>    If `odc-stac`_ is installed, you may specify `driver='rio'` to access the new hyperspectral loading and
>    dask-chunking features of ``stac_load``.

Reviewer: ```suggestion
   Use of this feature currently requires installing `odc-loader`_.

   If `odc-loader`_ is installed, you may specify `driver='rio'` or `driver='zarr'` when loading to access the new hyperspectral loading and
   dask-chunking features.
```

I probably should add an installation extra too...

7. The long-deprecated "ingestion" workflow and "executor" API have both been removed.

8. Multiple locations per dataset is now deprecated, and is not supported by the ``postgis`` index driver.

.. _`odc-geo`: https://github.com/opendatacube/odc-geo

> .. _`odc-stac`: https://github.com/opendatacube/odc-stac

Reviewer: ```suggestion
.. _`odc-loader`: https://github.com/opendatacube/odc-loader
```
